src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/csip/CSIPSynchronisationService.kt

```diff
@@ -38,59 +39,43 @@ class CSIPSynchronisationService(
       return
     }
 
+    val telemetry = mutableMapOf(
+      "nomisCSIPId" to event.csipReportId,
+      "offenderNo" to event.offenderIdDisplay,
+    )
+
     val nomisCSIP = nomisApiService.getCSIP(event.csipReportId)
-    mappingApiService.getCSIPReportByNomisId(
-      nomisCSIPReportId = event.csipReportId,
-    )?.let {
-      telemetryClient.trackEvent(
-        "csip-synchronisation-created-ignored",
-        event.toTelemetryProperties(it.dpsCSIPId),
-      )
-    } ?: let {
-      csipService.createCSIPReport(event.offenderIdDisplay, nomisCSIP.toDPSCreateRequest(), nomisCSIP.createdBy)
-        .also { dpsCsip ->
-          tryToCreateCSIPReportMapping(event, dpsCsip.recordUuid.toString()).also { result ->
-            telemetryClient.trackEvent(
-              "csip-synchronisation-created-success",
-              event.toTelemetryProperties(
-                dpsCsip.recordUuid.toString(),
-                result == MappingResponse.MAPPING_FAILED,
-              ),
-            )
+    mappingApiService.getCSIPReportByNomisId(nomisCSIPReportId = event.csipReportId)
+      ?.let {
+        // TODO CSIP Report mapping exists so we just want to update
+        //  csipService.syncCSIP(nomisCSIP.toDPSSyncRequest(it.dpsCSIPId), nomisCSIP.createdBy)
+        telemetryClient.trackEvent(
+          "csip-synchronisation-created-ignored",
+          event.toTelemetryProperties(it.dpsCSIPId),
+        )
+      }
+      ?: let {
+        // CSIP Report mapping doesn't exist
+        csipService.syncCSIP(nomisCSIP.toDPSSyncRequest(), nomisCSIP.createdBy)
+          .also { syncResponse ->
+            // At this point we need to determine all mappings and call the appropriate mapping endpoints
+            // For now, just map top level report
+            val csipReport = syncResponse.mappings.first { it.component == ResponseMapping.Component.RECORD }
+            val dpsCSIPReportId = csipReport.uuid.toString()
+
+            tryToCreateCSIPReportMapping(event, dpsCSIPReportId).also { result ->
+              val mappingSuccessTelemetry =
+                (if (result == MappingResponse.MAPPING_CREATED) mapOf() else mapOf("mapping" to "initial-failure"))
+              telemetryClient.trackEvent(
+                "csip-synchronisation-created-success",
+                telemetry + ("dpsCSIPId" to dpsCSIPReportId) + mappingSuccessTelemetry,
+              )
+            }
+            // TODO **** CHECK IF CHILD TABLES NEED MAPPING  ****
```

Comment:
Still lots of code to add in here to map child tables - this is just the first step, adding the top level parent with the new endpoint call